### PR TITLE
Wallet: implement 'Connecting' status, add 'disconnected' property

### PR DESCRIPTION
**Pull request by xiphon**
Closes https://github.com/monero-project/monero/issues/6308

monero-gui-connecting-status

### CODE REVIEW

File: components/NetworkStatusItem.qml [hunk 1]
```diff
@@ -39,23 +39,25 @@ Rectangle {
...
-            return qsTr("Wrong version")
-        if (status == Wallet.ConnectionStatus_Disconnected){
-            if(appWindow.walletMode <= 1){
-                return qsTr("Searching node") + translationManager.emptyString;
-            }
-            return qsTr("Disconnected")
+        switch (status) {
+            case Wallet.ConnectionStatus_Connected:
+                if (!appWindow.daemonSynced)
+                    return qsTr("Synchronizing");
```
Comment (rating89us) [inline components/NetworkStatusItem.qml]:
```suggestion
                    return qsTr("Synchronizing...");
```

File: components/NetworkStatusItem.qml [hunk 2]
```diff
@@ -39,23 +39,25 @@ Rectangle {
...
+                return appWindow.isMining ? qsTr("Connected") + " + " + qsTr("Mining"): qsTr("Connected");
+            case Wallet.ConnectionStatus_WrongVersion:
+                return qsTr("Wrong version");
+            case Wallet.ConnectionStatus_Disconnected:
+                if (appWindow.walletMode <= 1) {
+                    return qsTr("Searching node") + translationManager.emptyString;
+                }
+                return qsTr("Disconnected");
+            case Wallet.ConnectionStatus_Connecting:
+                return qsTr("Connecting");
```
Comment (rating89us) [inline components/NetworkStatusItem.qml]:
```suggestion
                return qsTr("Connecting...");
```

File: components/NetworkStatusItem.qml [hunk 3]
```diff
@@ -39,23 +39,25 @@ Rectangle {
...
+                if (!appWindow.daemonSynced)
+                    return qsTr("Synchronizing");
+                if (persistentSettings.useRemoteNode)
+                    return qsTr("Remote node");
+                return appWindow.isMining ? qsTr("Connected") + " + " + qsTr("Mining"): qsTr("Connected");
+            case Wallet.ConnectionStatus_WrongVersion:
+                return qsTr("Wrong version");
+            case Wallet.ConnectionStatus_Disconnected:
+                if (appWindow.walletMode <= 1) {
+                    return qsTr("Searching node") + translationManager.emptyString;
```
Comment (rating89us) [inline components/NetworkStatusItem.qml]:
```suggestion
                    return qsTr("Searching node...") + translationManager.emptyString;
```

File: components/NetworkStatusItem.qml (hunk 1 above)
Comment (xiphon) [reply components/NetworkStatusItem.qml]: Is not related to the PR

File: components/NetworkStatusItem.qml (hunk 3 above)
Comment (xiphon) [reply components/NetworkStatusItem.qml]: Is not related to the PR

File: components/NetworkStatusItem.qml (hunk 2 above)
Comment (xiphon) [reply components/NetworkStatusItem.qml]: Would prefer to keep it as is

**Pull request merged**